src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,89 @@
+package com.google.sps;
+
+import com.google.sps.user.User;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserTest {
+    private static final String ID = "userid";
+    private static final String EMAIL = "user@gmail.com";
+    private static final String NAME = "username";
+    private static final String SELF_INTRODUCTION = "I am the user";
+    private static final String IMG_URL = "/img.com";
+
+    @Test
+    public void getId() {
+        User newUser = new User.Builder(ID, EMAIL).build();
+        Assert.assertEquals(newUser.getId(), ID);
+    }
+
+    @Test
+    public void getEmail() {
+        User newUser = new User.Builder(ID, EMAIL).build();
+        Assert.assertEquals(newUser.getEmail(), EMAIL);
+    }
+
+    @Test
+    public void getName() {
+        // User with no preset name.
+        User newUser = new User.Builder(ID, EMAIL).build();
+        Assert.assertEquals(newUser.getName(), null);
+
+        // User with preset selfIntroduction.
+        newUser = new User.Builder(ID, EMAIL).setName(NAME).build();
+        Assert.assertEquals(newUser.getName(), NAME);
+
+        // User with preset selfIntroduction, and other fields.
+        newUser = new User.Builder(ID, EMAIL).setPublicPortfolio().setName(NAME).addSelfIntroduction(SELF_INTRODUCTION).build();
+        Assert.assertEquals(newUser.getName(), NAME);
+    }
+
+    @Test
+    public void getSelfIntroduction() {
+        // User with no preset selfIntroduction.
+        User newUser = new User.Builder(ID, EMAIL).build();
```

Comment:
Maybe also add a test where no parameters is given inside the `User.Builder()` ?

Reply:
You mean no parameters for the constructor of User.Builder?
Given that I have one constructor defined, I can't use a default no-argument constructor for User.Builder

Reply:
okay